grpc-server/hardware.go

```diff
@@ -285,3 +288,24 @@ func (s *server) Delete(ctx context.Context, in *hardware.DeleteRequest) (*hardw
 
 	return &hardware.Empty{}, err
 }
+
+func (s *server) validateHardwareData(ctx context.Context, hw *hardware.Hardware) error {
+	interfaces := hw.GetNetwork().GetInterfaces()
+	for i := range hw.GetNetwork().GetInterfaces() {
+		data, _ := db.GetByMAC(ctx, s.db, interfaces[i].GetDhcp().GetMac())
+		if data != "" {
+			logger.With("MAC", interfaces[i].GetDhcp().GetMac()).Info("Duplicate MAC address found")
+			newhw := hardware.Hardware{}
+			err := json.Unmarshal([]byte(data), &newhw)
+			if err != nil {
+				logger.Error(err, "Failed to unmarshal hardware data")
+				return err
+			}
+			if newhw.Id == hw.Id {
+				return nil
+			}
+			return errors.New("conflicting hardware MAC address " + interfaces[i].GetDhcp().GetMac() + " provided with hardware data/info")
```

Comment:
Please move all the error messages (across the file) to the top as constants. This is a practice we should follow going forward; no more inline messages. 
Eg:
```
const (
    conflictingMAC = "conflicting hardware MAC address %v provided with hardware data/info"
)

// you can now use it as 
return errors.New(fmt.Errorf(conflictingMAC, interfaces[i].GetDhcp().GetMac()))
```

Reply:
To move all the error messages across the file is a bit tidy at this moment. And also we need to do this for all the files so may that can be done in a seperate issue but for my changes I will follow this practise.